app/src/src_nonfreenet/org/breezyweather/sources/meteoam/MeteoAmService.kt

```diff
@@ -0,0 +1,100 @@
+import org.breezyweather.common.source.MainWeatherSource
+import org.breezyweather.common.source.ReverseGeocodingSource
+import org.breezyweather.common.source.SecondaryWeatherSourceFeature
+import org.breezyweather.sources.meteoam.json.MeteoAmForecastResult
+import org.breezyweather.sources.meteoam.json.MeteoAmObservationResult
+import org.breezyweather.sources.meteoam.json.MeteoAmReverseLocationResult
+import retrofit2.Retrofit
+
+class MeteoAmService @Inject constructor(
+    @ApplicationContext context: Context,
+    client: Retrofit.Builder
+) : HttpSource(), MainWeatherSource, ReverseGeocodingSource {
+
+    override val id = "meteoam"
+    override val name = "Servizio Meteo AM"
+    override val privacyPolicyUrl = "https://www.meteoam.it/it/privacy-policy"
+
+    override val color = Color.rgb(20, 122, 179)
+    // Required wording for third-party use taken from https://www.meteoam.it/it/condizioni-utilizzo
+    override val weatherAttribution = "Servizio Meteorologico dell’Aeronautica Militare. Informazioni elaborate utilizzando, tra l’altro, dati e prodotti del Servizio Meteorologico dell’Aeronautica Militare pubblicati sul sito www.meteoam.it"
+
+    private val mApi by lazy {
+        client
+            .baseUrl(METEOAM_BASE_URL)
+            .build()
+            .create(MeteoAmApi::class.java)
+    }
+
+    override val supportedFeaturesInMain = listOf<SecondaryWeatherSourceFeature>()
+
+    @SuppressLint("CheckResult")
+    override fun requestWeather(
+        context: Context, location: Location, ignoreFeatures: List<SecondaryWeatherSourceFeature>
+    ): Observable<WeatherWrapper> {
+        val forecast = mApi.getForecast(
+            location.latitude,
+            location.longitude
+        )
+        val observation = mApi.getCurrent(
+            location.latitude,
+            location.longitude
+        )
+        return Observable.zip(forecast, observation) {
+            forecastResult: MeteoAmForecastResult,
+            observationResult: MeteoAmObservationResult
+            ->
+            convert(forecastResult, observationResult)
+        }
+    }
+
+    // Reverse geocoding
+    override fun requestReverseGeocodingLocation(
+        context: Context,
+        location: Location
+    ): Observable<List<Location>> {
+        val reverseLocation = mApi.getReverseLocation(location.latitude, location.longitude)
+        val forecast = mApi.getForecast(location.latitude, location.longitude)
+        val locationList = mutableListOf<Location>()
+        return Observable.zip(reverseLocation, forecast) {
+            reverseLocationResult: MeteoAmReverseLocationResult,
+            forecastResult: MeteoAmForecastResult
+            ->
+            if (!reverseLocationResult.results.isNullOrEmpty() && !forecastResult.extrainfo?.timezone.isNullOrEmpty()) {
+                locationList.add(convert(location, reverseLocationResult.results[0], forecastResult.extrainfo?.timezone!!))
```

Comment:
Please make the `timezone` field not null rather than forcing Null Pointer Exceptions